--- ./extern/ros/sick_scan_xd/CMakeLists.txt.bak	2022-06-24 15:22:21.972634324 +0200
+++ ./extern/ros/sick_scan_xd/CMakeLists.txt	2022-06-24 15:25:23.206323867 +0200
@@ -566,9 +566,12 @@
         ${LDMRS_TARGET_DEPENDENCIES}
     )
 
-    rosidl_target_interfaces(sick_scan_lib ${PROJECT_NAME} "rosidl_typesupport_cpp")
-    rosidl_target_interfaces(sick_scan_shared_lib ${PROJECT_NAME} "rosidl_typesupport_cpp")
-    # rosidl_target_interfaces(sick_generic_caller ${PROJECT_NAME} "rosidl_typesupport_cpp")
+    rosidl_get_typesupport_target(sick_scan_lib_typesupport_target ${PROJECT_NAME} "rosidl_typesupport_cpp")
+    target_link_libraries(sick_scan_lib ${sick_scan_lib_typesupport_target})
+    rosidl_get_typesupport_target(sick_scan_shared_lib_typesupport_target ${PROJECT_NAME} "rosidl_typesupport_cpp")
+    target_link_libraries(sick_scan_shared_lib ${sick_scan_shared_lib_typesupport_target})
+    # rosidl_get_typesupport_target(sick_generic_caller_typesupport_target ${PROJECT_NAME} "rosidl_typesupport_cpp")
+    # target_link_libraries(sick_generic_caller ${sick_generic_caller_typesupport_target})
         
 endif(ROS_VERSION EQUAL 2)
 
