--- /opt/ros/galactic/lib/python3.8/site-packages/rclpy/node.py	2022-04-28 18:07:55.000000000 +0200
+++ /home/cris/Desktop/engineering-thesis-galactic/node.py	2022-05-26 03:09:39.657631809 +0200
@@ -205,12 +205,16 @@
         if parameter_overrides is not None:
             self._parameter_overrides.update({p.name: p for p in parameter_overrides})
 
+        # --------------------------------------------------------------------------
+        # @brief This section has been changed according to [2] in anticipation of
+        #    ROS2-galactic system patch
+        #
+        # @see [1] https://github.com/ros2/rclpy/issues/942
+        # @see [2] https://github.com/ros2/rclpy/commit/2c9ad988b277439d6faf41edf1cb28519e025fb5
+        # --------------------------------------------------------------------------
+
         # Clock that has support for ROS time.
-        # Note: parameter overrides and parameter event publisher need to be ready at this point
-        # to be able to declare 'use_sim_time' if it was not declared yet.
         self._clock = ROSClock()
-        self._time_source = TimeSource(node=self)
-        self._time_source.attach_clock(self._clock)
 
         if automatically_declare_parameters_from_overrides:
             self.declare_parameters(
@@ -221,6 +225,12 @@
                 ignore_override=True,
             )
 
+        # Init a time source.
+        # Note: parameter overrides and parameter event publisher need to be ready at this point
+        # to be able to declare 'use_sim_time' if it was not declared yet.
+        self._time_source = TimeSource(node=self)
+        self._time_source.attach_clock(self._clock)
+        
         if start_parameter_services:
             self._parameter_service = ParameterService(self)
 
